tests/linearizability/model.go

```diff
@@ -71,39 +102,79 @@ var etcdModel = porcupine.Model{
-			} else {
-				return fmt.Sprintf("put(%q, %q) -> ok, rev: %d", request.Key, request.PutData, response.Revision)
-			}
-		case Delete:
-			if response.Err != nil {
-				return fmt.Sprintf("delete(%q) -> %s", request.Key, response.Err)
-			} else {
-				return fmt.Sprintf("delete(%q) -> ok, rev: %d deleted:%d", request.Key, response.Revision, response.Deleted)
-			}
-		case Txn:
-			if response.Err != nil {
-				return fmt.Sprintf("txn(if(value(%q)=%q).then(put(%q, %q)) -> %s", request.Key, request.TxnExpectData, request.Key, request.TxnNewData, response.Err)
-			} else {
-				return fmt.Sprintf("txn(if(value(%q)=%q).then(put(%q, %q)) -> %v, rev: %d", request.Key, request.TxnExpectData, request.Key, request.TxnNewData, response.TxnSucceeded, response.Revision)
-			}
-		default:
-			return "<invalid>"
-		}
+		return describeEtcdRequestResponse(in.(EtcdRequest), out.(EtcdResponse))
 	},
 }
 
+func describeEtcdRequestResponse(request EtcdRequest, response EtcdResponse) string {
+	prefix := describeEtcdOperations(request.Ops)
+	if len(request.Conds) != 0 {
+		prefix = fmt.Sprintf("if(%s).then(%s)", describeEtcdConditions(request.Conds), prefix)
+	}
+
+	return fmt.Sprintf("%s -> %s", prefix, describeEtcdResponse(request.Ops, response))
+}
+
+func describeEtcdConditions(conds []EtcdCondition) string {
+	opsDescription := make([]string, len(conds))
+	for i := range conds {
+		opsDescription[i] = fmt.Sprintf("%s==%q", conds[i].Key, conds[i].ExpectedValue)
+	}
+	return strings.Join(opsDescription, ", ")
+}
+
+func describeEtcdOperations(ops []EtcdOperation) string {
+	opsDescription := make([]string, len(ops))
+	for i := range ops {
+		opsDescription[i] = describeEtcdOperation(ops[i])
+	}
+	return strings.Join(opsDescription, ", ")
+}
+
+func describeEtcdResponse(ops []EtcdOperation, response EtcdResponse) string {
+	if response.Err != nil {
+		return fmt.Sprintf("err: %q", response.Err)
+	}
+	if response.TxnFailure {
+		return fmt.Sprintf("txn failed, rev: %d", response.Revision)
+	}
+	respDescription := make([]string, len(response.Result))
+	for i := range response.Result {
+		respDescription[i] = describeEtcdOperationResponse(ops[i].Type, response.Result[i])
+	}
+	respDescription = append(respDescription, fmt.Sprintf("rev: %d", response.Revision))
+	return strings.Join(respDescription, ", ")
+}
+
+func describeEtcdOperation(op EtcdOperation) string {
+	switch op.Type {
+	case Get:
+		return fmt.Sprintf("get(%q)", op.Key)
+	case Put:
+		return fmt.Sprintf("put(%q, %q)", op.Key, op.Value)
+	case Delete:
+		return fmt.Sprintf("delete(%q)", op.Key)
+	default:
+		return "<invalid>"
+	}
+}
+
+func describeEtcdOperationResponse(op OperationType, resp EtcdOperationResult) string {
+	switch op {
+	case Get:
+		if resp.Value == "" {
+			return "nil"
+		}
+		return fmt.Sprintf("%q", resp.Value)
+	case Put:
+		return fmt.Sprintf("ok")
+	case Delete:
+		return fmt.Sprintf("deleted: %d", resp.Deleted)
+	default:
+		return "<invalid>"
```

Comment:
It might be misinterpreted as expected thing. 

`<! unknown op: %d !>`

Reply:
Done